### PR TITLE
Task #23 (test for idValidation middleware)

### LINKED ISSUE #23: Write unit test for idValidation middleware

### PR DESCRIPTION
**Pull request by SeredenkoDS**
Run the test using 'npx jest src/test/unit/middlewares/idValidation.spec.js'

### CODE REVIEW

**Review by MrAK-dev — APPROVED** [review]

**Review by olehpona — APPROVED** [review]

File: src/test/unit/middlewares/idValidation.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose')
+const { INVALID_ID } = require('~/consts/errors')
+const { createError } = require('~/utils/errorsHelper')
```
Comment (vlad-khrychov) [inline src/test/unit/middlewares/idValidation.spec.js]: I think better way will to mock `createError` with some implementation. And check what obtains this function and that it was called.

File: src/test/unit/middlewares/idValidation.spec.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose')
+const { INVALID_ID } = require('~/consts/errors')
```
Comment (vlad-khrychov) [inline src/test/unit/middlewares/idValidation.spec.js:2]: If you mock `createError` you could delete this import.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

File: src/test/unit/middlewares/idValidation.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+const idValidation = require('~/middlewares/idValidation')
+const { createError } = require('~/utils/errorsHelper')
+
+jest.mock('~/utils/errorsHelper', () => ({
+  createError: jest.fn((status, errorInfo) => {
+    return new Error();
```
Comment (vlad-khrychov) [inline src/test/unit/middlewares/idValidation.spec.js]:
```suggestion
    return new Error("test error");
```

File: src/test/unit/middlewares/idValidation.spec.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  it('should throw an error if id is invalid', () => {
+    const invalidId = 'invalid-id'
+
+    try {
+      idValidation(req, res, next, invalidId);
+    } catch (error) {
+      expect(createError).toHaveBeenCalledWith(400, INVALID_ID);
+    }
+    expect(next).not.toHaveBeenCalled()
```
Comment (vlad-khrychov) [inline src/test/unit/middlewares/idValidation.spec.js]:
```suggestion
    } catch (error) {
     expect(error.message).toEqual("test error")
    }
    expect(createError).toHaveBeenCalledWith(400, INVALID_ID);
    expect(next).not.toHaveBeenCalled()
```

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

**Review by tufned — APPROVED** [review]
There are a few semicolons
You can use  _prettier_ to format file
Overall looks good)

**Review by HalynaProlubshchykova — APPROVED** [review]

**Comment by SeredenkoDS** [comment]
Thank you for the comments! Made the suggested changes.

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**